### PR TITLE
Initial draft for GCConserveMemory config.

### PR DESCRIPTION
**Pull request by PeterSolMS**
## Summary

Describes the GCConserveMemory setting.

### CODE REVIEW

File: docs/core/runtime-config/garbage-collector.md [hunk 1]
```diff
@@ -494,3 +494,33 @@ Example:
...
+| | Setting name | Values | Version introduced |
+| - | - | - | - |
+| **runtimeconfig.json** | N/A | N/A | N/A |
+| **Environment variable** | `COMPlus_GCConserveMemory` | `0` -`9` | .NET Framework 4.8 |
+| **Environment variable** | `COMPlus_GCConserveMemory` | `0` -`9` | .NET 6 |
+| **app.config for .NET Framework** | `GCConserveMemory` | `0` -`9` | .NET Framework 4.8 |
+
+Example:
+
+```app.config file
```
Comment (Youssef1313) [inline docs/core/runtime-config/garbage-collector.md:514]:
~~~suggestion
```xml
~~~

File: docs/core/runtime-config/garbage-collector.md [hunk 2]
```diff
@@ -494,3 +494,33 @@ Example:
...
+
+<configuration>
+  <runtime>
+    <GCConserveMemory  enabled="5"/>
```
Comment (Youssef1313) [inline docs/core/runtime-config/garbage-collector.md:518]:
```suggestion
    <GCConserveMemory enabled="5"/>
```

File: docs/core/runtime-config/garbage-collector.md [hunk 3]
```diff
@@ -494,3 +494,33 @@ Example:
...
+  </runtime>
+</configuration>
+
```
Comment (Youssef1313) [inline docs/core/runtime-config/garbage-collector.md:521]:
```suggestion
```

File: docs/core/runtime-config/garbage-collector.md [hunk 4]
```diff
@@ -494,3 +494,33 @@ Example:
...
+```
+> [!TIP]
+> Experiment with different numbers to see which value works best for you.
```
Comment (gewarren) [inline docs/core/runtime-config/garbage-collector.md:524]:
```suggestion
> Experiment with different numbers to see which value works best for you. Start with a value between 5 and 7.
```

File: docs/core/runtime-config/garbage-collector.md [hunk 5]
```diff
@@ -494,3 +494,33 @@ Example:
...
+> Usually we start with a value between 5 and 7.
```
Comment (gewarren) [inline docs/core/runtime-config/garbage-collector.md:525]:
```suggestion
```

File: docs/core/runtime-config/garbage-collector.md [hunk 6]
```diff
@@ -494,3 +494,33 @@ Example:
...
+- Configures the garbage collector to conserve memory at the expense of more frequent garbage collections and possibly longer pause times.
+- Default value is 0 - this implies no change.
+- Besides the default value 0, values between 1 and 9 (inclusive) are valid. The higher the value, the more the garbage collector tries to conserve memory and thus to keep the heap small.
+- If the value is non-zero, the large object heap will be compacted automatically if it has too much fragmentation.
+
+| | Setting name | Values | Version introduced |
+| - | - | - | - |
+| **runtimeconfig.json** | N/A | N/A | N/A |
+| **Environment variable** | `COMPlus_GCConserveMemory` | `0` -`9` | .NET Framework 4.8 |
+| **Environment variable** | `COMPlus_GCConserveMemory` | `0` -`9` | .NET 6 |
```
Comment (gewarren) [inline docs/core/runtime-config/garbage-collector.md:509]:
```suggestion
| **Environment variable** | `DOTNET_GCConserveMemory` | `0` -`9` | .NET 6 |
```

**Review by gewarren — COMMENTED** [review]
Thank you @PeterSolMS ! This is great.

**Review by Maoni0 — APPROVED** [review]

File: docs/core/runtime-config/garbage-collector.md (hunk 1 above)
Comment (PeterSolMS) [reply docs/core/runtime-config/garbage-collector.md:514]: Ok.

File: docs/core/runtime-config/garbage-collector.md (hunk 2 above)
Comment (PeterSolMS) [reply docs/core/runtime-config/garbage-collector.md:518]: Thanks for catching!

File: docs/core/runtime-config/garbage-collector.md (hunk 4 above)
Comment (PeterSolMS) [reply docs/core/runtime-config/garbage-collector.md:524]: That's better.

File: docs/core/runtime-config/garbage-collector.md (hunk 6 above)
Comment (PeterSolMS) [reply docs/core/runtime-config/garbage-collector.md:509]: Sure, DOTNET_GCConserveMemory certainly works, COMPlus_GCConserveMemory works as well. Put both or just the former one?

Comment (gewarren) [reply docs/core/runtime-config/garbage-collector.md]: To be consistent with the other settings, we should list both. Thanks!

File: docs/core/runtime-config/garbage-collector.md [hunk 7]
```diff
@@ -494,3 +494,31 @@ Example:
...
+| **Environment variable** | `DOTNET_GCConserveMemory` | `0` -`9` | .NET 6 |
+| **app.config for .NET Framework** | `GCConserveMemory` | `0` -`9` | .NET Framework 4.8 |
+
+Example:
+
+```xml
+
+<configuration>
+  <runtime>
+    <GCConserveMemory enabled="5"/>
```
Comment (gewarren) [inline docs/core/runtime-config/garbage-collector.md:518]: Looks like we need to add a page for this element here as well: https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/. Let me know if you need help with that.

File: docs/core/runtime-config/garbage-collector.md (hunk 6 above)
Comment (PeterSolMS) [reply docs/core/runtime-config/garbage-collector.md]: Will do.

File: docs/core/runtime-config/garbage-collector.md [hunk 8]
```diff
@@ -494,3 +494,31 @@ Example:
...
+- If the value is non-zero, the large object heap will be compacted automatically if it has too much fragmentation.
+
+| | Setting name | Values | Version introduced |
+| - | - | - | - |
+| **runtimeconfig.json** | N/A | N/A | N/A |
+| **Environment variable** | `COMPlus_GCConserveMemory` | `0` -`9` | .NET Framework 4.8 |
+| **Environment variable** | `DOTNET_GCConserveMemory` | `0` -`9` | .NET 6 |
+| **app.config for .NET Framework** | `GCConserveMemory` | `0` -`9` | .NET Framework 4.8 |
+
+Example:
```
[the remark this answers is not in the document]
Comment (PeterSolMS) [reply docs/core/runtime-config/garbage-collector.md:512]: Sure.

File: docs/core/runtime-config/garbage-collector.md (hunk 7 above)
Comment (PeterSolMS) [reply docs/core/runtime-config/garbage-collector.md:519]: Will do. Should it be part of this PR?

Comment (PeterSolMS) [reply docs/core/runtime-config/garbage-collector.md:519]: I'll do it separately, so I'll merge this PR now.

**Pull request merged**